src/reader.rs

```diff
@@ -535,6 +550,17 @@ impl<T: Read + Seek> Reader<T> {
         Ok(())
     }
 
+    /// Returns the number of the shapes in the shapefile
+    ///
+    /// # Error
+    ///
+    /// Returns [Error::MissingIndexFile] if the _shx_ file
+    /// was not found by [Reader::from_path], or [ShapeReader]
+    /// given in [Reader::new] was created without index.
```

Comment:
I think here, to avoid duplication we could just put

```Rust
/// Returns the number of shapes in the shapefile
/// 
/// See [ShapeReader::shape_count]
````